controllers/resource/template.go

```diff
@@ -175,6 +175,7 @@ func (r *CloudStackTemplate) TemplateResources(ctx context.Context, eksaCluster
 	for _, wnConfig := range clusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations {
 		workerNodeGroupMachineSpecs[wnConfig.MachineGroupRef.Name] = workerCsmcs[wnConfig.MachineGroupRef.Name].Spec
 	}
+	clusterSpec.CloudStackDatacenter = &csdc
```

Comment:
That's what I've been going back and forth with @g-gaston about. It seems like the CloudStackDatacenter is never present when we build the clusterSpec and I don't see why it would be. The code to build the clusterSpec is provider agnostic, so I'm not sure where the CloudStackDatacenter would be set, unless it has something to do with Config Processors/ConfigManagerEntry stuff? I'm not familiar with those pieces.

Reply:
I think the reason is we build the spec in controller by https://github.com/aws/eks-anywhere/blob/main/pkg/cluster/spec.go#L262 which does not update `Config.cloudstackdatacenter` object. The build spec method used in CLI is https://github.com/aws/eks-anywhere/blob/main/pkg/cluster/spec.go#L184 where we use config manager and properly set up the provider spec.

Reply:
https://github.com/aws/eks-anywhere/issues/2929